pkg/skaffold/build/gcb/desc.go

```diff
@@ -24,48 +24,40 @@ import (
 	cloudbuild "google.golang.org/api/cloudbuild/v1"
 )
 
-func (b *Builder) buildDescription(artifact *latest.Artifact, tag, bucket, object string) (*cloudbuild.Build, error) {
-	tags := []string{tag}
-
-	steps, err := b.buildSteps(artifact, tag)
-	if err != nil {
-		return nil, err
-	}
-
-	return &cloudbuild.Build{
+func (b *Builder) baseDescription(bucket, object string) cloudbuild.Build {
+	return cloudbuild.Build{
 		LogsBucket: bucket,
 		Source: &cloudbuild.Source{
 			StorageSource: &cloudbuild.StorageSource{
 				Bucket: bucket,
 				Object: object,
 			},
 		},
-		Steps:  steps,
-		Images: tags,
 		Options: &cloudbuild.BuildOptions{
 			DiskSizeGb:  b.DiskSizeGb,
 			MachineType: b.MachineType,
 		},
 		Timeout: b.Timeout,
-	}, nil
+	}
 }
 
-func (b *Builder) buildSteps(artifact *latest.Artifact, tag string) ([]*cloudbuild.BuildStep, error) {
+func (b *Builder) buildDescription(artifact *latest.Artifact, tag, bucket, object string) (cloudbuild.Build, error) {
+	base := b.baseDescription(bucket, object)
+
 	switch {
 	case artifact.DockerArtifact != nil:
-		return b.dockerBuildSteps(artifact.DockerArtifact, tag)
+		return b.dockerBuildDescription(artifact.DockerArtifact, tag, base)
```

Comment:
I found this passing `base` a bit puzzling, especially since it wasn't being passed as a pointer.  You're now turning these `xxxBuildSteps()` methods into something that re-casts or morphs the `base` instance for that builder.  Perhaps moving the `cloudbuild.Builder` argument to be first, so that it's a bit more clear that the methods are transforming or rewriting the instance?

(Or could the `xxxBuildDescription()` methods just call `baseDescription` themselves?)